source/adapters/hip/device.cpp

```diff
@@ -724,6 +724,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
   }
 
   case UR_DEVICE_INFO_GLOBAL_MEM_FREE: {
+    // Work around an issue on some (unsupported) architectures,
+    // where hipMemGetInfo fails internally and returns hipErrorInvalidValue
+    // when trying to query the amount of available global memory. Since we
+    // can't distinguish this condition from us doing something wrong, we can't
+    // handle it gracefully.
+    hipDeviceProp_t Props;
+    detail::ur::assertion(hipGetDeviceProperties(&Props, hDevice->get()) ==
```

Comment:
nit: instead of using assert atm all hip calls like this should ideally be wrapped in `UR_CHECK_ERROR(` that at least points to the user to the problem code if there is an error.

Reply:
Think this is probably especially important with things like hip where random errors come up all the time for certain devices.

Reply:
Techically both ur::assertion and UR_CHECK_ERROR are not consistent with sycl spec that expects runtime errors for such failures, but I think UR_CHECK_ERROR provides at least more info for the user.

Reply:
I'll approve it anyway because trying to enforce a consistent way of dealing with error passing has proven impossible!

Reply:
Interesting, thanks for the in-depth explanation. Yeah I was just copying nearby code but I see that `UR_CHECK_ERROR` is used sporadically too. I will switch to `UR_CHECK_ERROR`. I see it does logging, which also sounds nice.